pkg/controller/servicebindingrequest/volumemount/retriever.go

```diff
@@ -0,0 +1,198 @@
+package volumemount
+
+import (
+	"context"
+	"fmt"
+	"strings"
+
+	"github.com/go-logr/logr"
+	"github.com/redhat-developer/service-binding-operator/pkg/controller/servicebindingrequest/planner"
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/types"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	logf "sigs.k8s.io/controller-runtime/pkg/runtime/log"
+)
+
+// Retriever reads all data referred in plan instance, and store in a secret.
+type Retriever struct {
+	ctx    context.Context   // request context
+	client client.Client     // Kubernetes API client
+	plan   *planner.Plan     // plan instance
+	logger logr.Logger       // logger instance
+	Data   map[string][]byte // data retrieved
+}
+
+// Data contains the data to mount
+type Data struct {
+	Prefix string
+	Value  []byte
+}
+
+const (
+	bindingPrefix = "SERVICE_BINDING"
+	secretPrefix  = "urn:alm:descriptor:servicebindingrequest:volumemount:secret"
+)
+
+// getNestedValue retrieve value from dotted key path
+func (r *Retriever) getNestedValue(key string, sectionMap interface{}) (string, error) {
+	if !strings.Contains(key, ".") {
+		value, exists := sectionMap.(map[string]interface{})[key]
+		if !exists {
+			return "", fmt.Errorf("Can't find key '%s'", key)
+		}
+		return fmt.Sprintf("%v", value), nil
+	}
+	attrs := strings.SplitN(key, ".", 2)
+	newSectionMap, exists := sectionMap.(map[string]interface{})[attrs[0]]
+	if !exists {
+		return "", fmt.Errorf("Can't find '%v' section in CR", attrs)
+	}
+	return r.getNestedValue(attrs[1], newSectionMap.(map[string]interface{}))
+}
+
+// getCRKey retrieve key in section from CR object, part of the "plan" instance.
+func (r *Retriever) getCRKey(section string, key string) (string, error) {
```

Comment:
Isn't everything here a duplicate of `pkg/controller/servicebindingrequest/retriever.go`  ?

Reply:
Yes, but here a new intermediary secret object is being created

Reply:
Can we address this in a separate PR?

Reply:
Sure!